cli/internal/terraform/terraform/azure/main.tf

```diff
@@ -44,7 +44,7 @@ resource "random_password" "initSecret" {
 resource "azurerm_attestation_provider" "attestation_provider" {
   count = var.create_maa ? 1 : 0
   # name must be between 3 and 24 characters in length and use numbers and lower-case letters only.
-  name                = format("%sap", replace(var.name, "/[^a-z0-9]/", ""))
+  name                = format("%sap", replace(local.name, "/[^a-z0-9]/", ""))
```

Comment:
Constellation limits names on Azure (without uid suffix) to 37 characters.
However, MAA names are limited to only 24 characters. Since we also add `sap` and a uid suffix to the name, we can easily run into configurations that are accepted by our config validation, but will fail during creation with Terraform.

I would suggest using only the uid of a constellation to generate the MAA name.
Example: `constellXXXX`, where `XXXX` is the Constellation's UID